### ISSUE TITLE
Release SE 4.0.5

### ISSUE THREAD

**Issue by niksedk** [SubtitleEdit/subtitleedit#8143]
It's time for yet another release :)

Main changes:
- New auto-translate engines: [LM Studio](https://github.com/SubtitleEdit/subtitleedit/issues/7704), Ollama, Antrophic.
- New "Open video" toolbar button/image (not enabled by default - nice if you start your workflow with Whisper).
- Embed (SoftSub) in video file now supports PGS.
- Update: Purfview Faster Whisper/libmpv/ffmpeg/yt-dlp
- New "fast" options for PFW (single word, high-light spoken word) - see https://github.com/SubtitleEdit/subtitleedit/issues/8069 and https://github.com/SubtitleEdit/subtitleedit/discussions/8066
- New shortcut "Insert new subtitle at video pos (fill)" works well with audio to text - see https://github.com/SubtitleEdit/subtitleedit/issues/8109

Read more about fixes and improvements in the change log: 
https://raw.githubusercontent.com/SubtitleEdit/subtitleedit/master/Changelog.txt

Please download the beta and give a quick test-run:
https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.4/SubtitleEditBeta.zip

**Comment by Purfview** [comment]
There is Faster-Whisper `r192.1`.

**Comment by niksedk** [comment]
> There is Faster-Whisper `r192.1`.

thx - beta updated: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.4/SubtitleEditBeta.zip

**Comment by Purfview** [comment]
Oh, I see that I didn't remove all not actual checks for `--sentence` :

```
Only the alternative funtionality is allowed with `--sentence`.

Error: Original functionality of 'max_line_width' and 'max_line_count' parameters is borked.
```

"highlight_words" doesn't need `--sentence` and it actually doesn't do anything for `--highlight_words`.
I'll quickly compile corrected version.

**Comment by Purfview** [comment]
Released Faster-Whisper `r192.2`: Allowed to use 'max_line_width' and 'max_line_count' without '--sentence'.
And please remove `--sentence` from advanced settings 'Highlight current word' button.

**Comment by Purfview** [comment]
I don't see SE working on the original media, log shows it's working on 16Khz temp files.

https://github.com/SubtitleEdit/subtitleedit/issues/8097

**Comment by Purfview** [comment]
> It depends on how you start Whisper from SE... I normally load a video first, and then start whisper.
How do you start whisper from SE?

I select "Audio to text" from the menu, I dunno other methods to start whisper.

>Also, r192.2 seems to have broken --one_word 2 ?

Shouldn't be, I'll check.

**Comment by Purfview** [comment]
Released `r192.3`: Bugfix: 'one_word' was broken in r192.2.

**Comment by niksedk** [comment]
SE updated with latest version of Purfview's Faster Whisper: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.4/SubtitleEditBeta.zip

(Should now send video file to purfview faster whisper when using Video - Audio to text via Whisper)

**Comment by Purfview** [comment]
Now it works.
One issue left - when selecting not the first audio track SE could pass "--ff_track" arg.
"--ff_track 2" selects the second audio track.

**Comment by gericho** [comment]
Using `ollama serve` I got these

[GIN] 2024/04/08 - 22:57:08 | 200 |    855.8236ms |       127.0.0.1 | POST     "/api/chat"
[GIN] 2024/04/08 - 22:57:50 | 404 |            0s |       127.0.0.1 | POST     "/api/generate"

so the chat in a separated command prompt works, but SE does not, using gemma:7b and ollama latest.

I found this, if it could help [here](https://github.com/ollama/ollama/issues/2736)

Thank you!

**Comment by niksedk** [comment]
> Using `ollama serve` I got these...

thx :)

Should be fixed in latest beta - a model was needed (you can always check the `error_log.txt` file for more info when having an error)

Also, single lines should be used for this (at least for llama2), which you can enable by right-clicking:

![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/c8d2e726-e1a4-40dd-8fba-10868dbaeab0)

**Comment by gericho** [comment]
Thank you! I assume that the latest version downloads the model automatically

To solve it temporary (or if you may want to add to the FAQ):
1. install the server setup from [ollama.com](https://ollama.com/) (or better from the GitHub release page)
2. once server is running in tray, pull the model using a PowerShell with:
`ollama pull gemma:7b`

Select of course the same model pulled now in the SE, and then the translation may start normally.
Thank you!

**Comment by niksedk** [comment]
The single-line function only refers to that one subtitle line is translated in it's own call. Otherwise lines are merged to larger bulks.

Translate just runs from the selected line to the end...

**Comment by gericho** [comment]
Perhaps I'm being too demanding, as LLM models are non-deterministic, incorporating a 'Regenerate' button on the main interface for line-by-line translation regeneration would be beneficial.

**Comment by davepusey** [comment]
Can we make sure this gets pushed to NuGet too please, as the last available version on there is still 3.6.13...

See https://www.nuget.org/packages/libse

Thank you.

**Comment by niksedk** [comment]
> Can we make sure this gets pushed to NuGet too please, as the last available version on there is still 3.6.13...
> 
> See https://www.nuget.org/packages/libse
> 
> This is especially important to me as I can then starting making use of the changes from my recent pull request #8132 .
> 
> Thank you.

sure :)
